### ISSUE TITLE
Large multi-channel inverse FFTs give wrong results

### ISSUE THREAD

**Issue by eelregit** [google/jax#18453]
### Description

```python
import jax.numpy as jnp
from jax import random

key = random.PRNGKey(0)

def fun(key, length):
    size = (length,) * 3
    last3 = range(-3, 0)
    f = random.normal(key, (3,) + size)  # 3 channels
    f = jnp.fft.fftn(f, axes=last3) 

    # 1st way to do the same inverse FFT, all at once
    f_ = jnp.fft.ifftn(f, axes=last3)
    print(f_[0, 0, 0, 1])
    del f_

    # 2nd way to do the same inverse FFT, channel by channel
    f_ = jnp.stack([jnp.fft.ifftn(g) for g in f], axis=0)
    print(f_[0, 0, 0, 1])

fun(key, 890)  # prints (-0.93745315-9.771974e-08j) (-0.93745315-9.771974e-08j)
fun(key, 900)  # prints (-1235954700+99.51599j) (-1.6954112+1.9948014e-07j)
```

For 3D FFT with grid size <= 890, the two methods to do the same inverse FFT agree and give the correct results (expected to be or close to be real numbers), while for size >= 900, the 1st method gives the wrong result.

### What jax/jaxlib version are you using?

jax 0.4.19 jaxlib 0.4.19+cuda12.cudnn89

### Which accelerator(s) are you using?

GPU

### NVIDIA GPU info

```
Thu Nov  9 07:27:37 2023       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.86.10              Driver Version: 535.86.10    CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA H100 PCIe               Off | 00000000:19:00.0 Off |                    0 |
| N/A   40C    P0              73W / 350W |  80821MiB / 81559MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A    434548      C   ...ome/yinli/mamba/envs/jax/bin/python    80808MiB |
+---------------------------------------------------------------------------------------+
```

**Comment by hawkinsp** [comment]
My strong suspicion is that this is an integer overflow somewhere (in XLA or cufft), because:
```
In [1]: 890*890*890*3
Out[1]: 2114907000

In [2]: 900*900*900*3
Out[2]: 2187000000

In [3]: 2**31
Out[3]: 2147483648
```

**Comment by hawkinsp** [comment]
I'm pretty sure this is an integer overflow in cufft: it looks to me like XLA is passing reasonable parameters to cufft.

(Note: you need a 80GB GPU to reproduce this: it reproduces on H100 at least.)

**Comment by nouiz** [comment]
@eelregit I was able to reproduce on A100 with CUDA 12.2. 
Can you try with CUDA 12.3? I just tried it and I got good results.

**Comment by hawkinsp** [comment]
We conferred and we think this is a bug in JAX actually. Specifically, the normalization here: https://github.com/google/jax/blob/d488714689cb4373dc4ec54149c8746455c95397/jax/_src/numpy/fft.py#L37

will happen in int32 unless `JAX_ENABLE_X64=1` is set.

**Comment by eelregit** [comment]
Thanks! That makes sense. Maybe turn `s` into weak typed float before the `prod` can help?
